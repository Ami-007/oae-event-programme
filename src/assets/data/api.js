import axios from "axios";

export const getEvent = async (id) => {
  const url = `https://oae-mock-api.herokuapp.com/events/${id}`;

  return axios.get(url).then((event) => event.data);
};
