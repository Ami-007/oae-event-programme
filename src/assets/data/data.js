export const getMockEventById = (id) => {
  return events.filter((event) => event.id === id.toString())[0];
};

// MOCK DATA
export const events = [
  {
    id: "6bb8ffe8-6e5e-4abc-be7f-b05b749849cf",
    name: "Infant Musicality",
    series: "Bach, The Universe & Everything",
    time: "11.30",
    date: "12-12-21",
    venue: "King's Place",
    location: "London",
    imageSrc:
      "https://oae.co.uk/wp-content/uploads/2021/07/MicrosoftTeams-image-5-1250x825.jpg",
    imageAlt: "Event Poster",
    content: [
      {
        heading:
          "If this is your first Bach, the Universe & Everything (BAE), welcome!",
        text: "We like to think of the series as a community, similar to the one Bach enjoyed in Leipzig where he produced cantatas at an extraordinary rate, providing innovative music for the weekly services at the church where he worked from 1723 until his death in 1750. At that time, the congregation was as open to science and new ideas as it was united by faith. It is in the spirit that we come together for a thought-provoking cantata and a talk from a leading scientist. Professor Jon Chapman from the University of Oxford invites us to ponder the fleeting moment when sound waves reach our ears before they disappear. Using some careful calculations, Jon suggests that we might even be able to hear the shape of a drum.",
      },
      {
        heading:
          "A note from our Principal Keyboard, Steven Devine, about today's programme",
        text: "The Cantata being performed today is BWV 26 - Ach wie flüchtig, ach wie nichtig. The chorale of this name forms the linchpin of the outer movements and it is the only time Bach used this chorale in his cantatas. The melody of the chorale, written by Michael Franck in 1652, is used by others and the organ prelude today is a movement for manuals only (that is, no organ pedals!) from a set of variations on this tune by Georg Bohm (1661 - 1733), an organist whom J S Bach hugely admired and who may even have been his teacher.The irresistable combination of three oboes brought a wealth of different repertoire to mind, but I have been enjoying digging into lesser-known works by George-Philipp Telemann this season and this movement from his “Darmstadt” Ouverture (the title of which roughly translates as “boasting”!) is a witty interplay between the wind and string teams.",
      },
    ],
    schedule: [
      {
        name: "Welcome",
        title: null,
        author: null,
        description:
          "Non ad labore aliqua adipisicing et laborum exercitation. Enim consectetur aliquip eu commodo. Eiusmod ipsum anim exercitation consectetur elit amet voluptate aliquip ullamco. Duis deserunt dolore et officia ullamco voluptate culpa eiusmod cupidatat ullamco sit anim incididunt nulla. Nisi et et commodo non dolor amet aliquip exercitation. Et fugiat consequat mollit cillum anim aliquip commodo laboris enim voluptate ullamco ipsum nostrud qui. Laboris duis cillum duis aliqua cupidatat nulla.",
        alternateDescription:
          "Not to do any iterative fatigue and fatigue. For the main objectives of the EU. This type of exercise enhanced the mind's desire to enjoy the objectives of any school. They abandon the school district offices in pain and pleasure through no fault of that kind If not, the pain will not be comfortable and the objectives of the exercise are important. And he shuns the consequences of the softening of his soul, for the pleasure of his labors, for the pleasure of any one of us. Laboris duis cillum duis something longing for none.",
      },
      {
        name: "Cantata",
        title: "Ach wie fluchtig, ach wie nichtig BWV 26",
        author: "JS Bach",
        description:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id porta nibh venenatis cras sed felis eget velit.",
        alternateDescription:
          "The pain itself is the love of the pain, the main ecological problems, but I give this kind of time to fall down, so that some great pain and pain. That's the gate of the venenatis tomorrow, but the cat needs to choose.",
      },
    ],
    theme: {
      fontType: 1,
      primaryColour: "#8BA0E3",
      textColour: "#FFFFFF",
      highlightColour: "#FFFFC7",
    },
  },
  {
    id: "5101a840-5606-4097-9b66-a567213ae915",
    name: "Can You Hear The Shape of a Drum?",
    series: "Bach, The Universe & Everything",
    time: "11.30",
    date: "21-11-21",
    venue: "King's Place",
    location: "London",
    imageSrc:
      "https://oae.co.uk/wp-content/uploads/2021/07/Shape-of-drum-artwork-fb-event-1250x825.jpg",
    imageAlt: "Event Poster",
    featuredEvent: false,
    content: [
      {
        heading:
          "If this is your first Bach, the Universe & Everything (BAE), welcome!",
        text: "We like to think of the series as a community, similar to the one Bach enjoyed in Leipzig where he produced cantatas at an extraordinary rate, providing innovative music for the weekly services at the church where he worked from 1723 until his death in 1750. At that time, the congregation was as open to science and new ideas as it was united by faith. It is in the spirit that we come together for a thought-provoking cantata and a talk from a leading scientist. Professor Jon Chapman from the University of Oxford invites us to ponder the fleeting moment when sound waves reach our ears before they disappear. Using some careful calculations, Jon suggests that we might even be able to hear the shape of a drum.",
      },
      {
        heading: "Can you hear the shape of a Drum?",
        text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam sagittis augue et urna volutpat aliquam quis at ipsum. ",
      },
    ],
    schedule: [
      {
        name: "Welcome",
        title: null,
        author: null,
        description: null,
        alternateDescription: null,
      },
      {
        name: "Prelude",
        title: "Ach wie fluchtig, ach wie nichtig",
        author: "Georg Bohm",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Polyphony",
        title: "Haec Dies",
        author: "William Byrd",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Reading",
        title: "Beat! Beat! Drums!",
        author: "Walt Whitman",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Cantata",
        title: "Ach wie fluchtig, ach wie nichtig BWV 26",
        author: "JS Bach",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Science",
        title: "Can You Hear The Shape of a Drum?",
        author: "Professor John Chapman",
        description: null,
        alternateDescription: null,
      },
      {
        name: "Closing",
        title: null,
        author: null,
        description: null,
        alternateDescription: null,
      },
      {
        name: "Postlude",
        title: "Darmstadt Ouverture TWV55:g4 'Gasconnade'",
        author: "Telemann",
        description: null,
        alternateDescription: null,
      },
    ],
    theme: {
      fontType: 2,
      primaryColour: "#FA825D",
      textColour: "#FFFFFF",
      highlightColour: "#FFFFC7",
    },
  },
];
