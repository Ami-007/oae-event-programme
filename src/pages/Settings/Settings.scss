* {
  margin-top: 0;
}

.setting-font-size {
  margin-top: 2rem;
  border: none;
  margin-right: 4rem;
  position: relative;
  left: 1.6rem;
  border-bottom: 1px solid #e4e5f0;
  &__header {
    font-weight: 700;
  }
  &__input {
    margin-left: 1rem;
  }
  &__label {
    margin-right: 1rem;
  }
}
