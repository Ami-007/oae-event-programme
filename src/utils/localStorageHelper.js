export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
