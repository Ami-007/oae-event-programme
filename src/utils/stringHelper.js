export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
