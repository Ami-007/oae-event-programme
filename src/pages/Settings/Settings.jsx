import React, { useState, useEffect } from "react";
import "./Settings.scss";
import Layout from "../../components/Layout/Layout";
import { useParams } from "react-router-dom";
import { getEvent } from "../../assets/data/api";
import LoadingSpinner from "../../components/LoadingSpinner/LoadingSpinner";
import { getDarkTheme, setDarkTheme, getFontSize, setFontSize } from "../../utils/localStorageHelper";
import ErrorPage from "../ErrorPage/ErrorPage";
import SettingsItemToggle from "../../components/SettingsItemToggle/SettingsItemToggle";
import PageHeader from "../../components/PageHeader/PageHeader";

const Settings = () => {
  const { eventId } = useParams();
  const [event, setEvent] = useState();

  useEffect(async () => {
    setEvent((await getEvent(eventId)) ?? null);
  }, [getEvent, eventId, setEvent]);

  if (event === undefined) return <LoadingSpinner />;

  if (event === null) return <ErrorPage />;

  const { theme } = event;

  const handleChange = (e) => {
    let checked = e.target.checked;
    setDarkTheme(checked);
    window.location.reload();
  };

  const isDarkMode = getDarkTheme();

  const fontSize = getFontSize();

  const handleClick = (num) => {
    setFontSize(num);
    window.location.reload();
  };

  const isInputChecked = (inputFontSize) => {
    if (fontSize === null) {
      return inputFontSize === "100%";
    }
    return inputFontSize === fontSize;
  };

  return (
    <Layout eventId={eventId} theme={theme.templateTheme.toLowerCase()} themeColorPrimary={theme.primaryColor} themeColorText={theme.textColour} themeColorHighlight={theme.subtitleColor}>
      <div className="settings-content">
        <PageHeader title="Settings" className="settings-content__header" />
        <SettingsItemToggle settingsLabel="Dark Mode" onChange={handleChange} checked={isDarkMode} />

        <fieldset className="setting-font-size">
          <legend className="setting-font-size__header">Font Size</legend>
          <input className="setting-font-size__input" type="radio" onClick={() => handleClick("100%")} click={fontSize} defaultChecked={isInputChecked("100%")} name="font-size" />
          <label className="setting-font-size__label" htmlFor="small-font">
            Small
          </label>
          <input className="setting-font-size__input" type="radio" onClick={() => handleClick("125%")} click={fontSize} defaultChecked={isInputChecked("125%")} name="font-size" />
          <label className="setting-font-size__label" htmlFor="medium-font">
            Medium
          </label>
          <input className="setting-font-size__input" type="radio" onClick={() => handleClick("150%")} click={fontSize} defaultChecked={isInputChecked("150%")} name="font-size" />
          <label className="setting-font-size__label" htmlFor="large-font">
            Large
          </label>
        </fieldset>
      </div>
    </Layout>
  );
};

export default Settings;
