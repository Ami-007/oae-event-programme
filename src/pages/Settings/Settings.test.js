import { customRender } from "../../utils/testUtils";
import Settings from "./Settings";
import { screen } from "@testing-library/react";
import { getMockEventById } from "../../assets/data/data";

const mockEvent = getMockEventById("6bb8ffe8-6e5e-4abc-be7f-b05b749849cf");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: () => [mockEvent, jest.fn()],
}));

test("renders settings component", () => {
  const { container } = customRender(<Settings />);
  expect(container).toMatchSnapshot();
});

test("renders settings component", () => {
  customRender(<Settings />);
  const inputElement = screen.getByRole("checkbox");
  expect(inputElement).toBeInTheDocument();
});
