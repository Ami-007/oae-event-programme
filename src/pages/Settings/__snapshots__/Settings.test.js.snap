// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders settings component 1`] = `
<div>
  <div
    class="layout modern"
  >
    <div>
      <div
        class="settings-content"
      >
        <div
          class="page-header page-header--only"
        >
          <div
            class="header-home"
          >
            <span
              class="header-home__subtitle"
            />
            <h1
              class="header-home__title"
            >
              Settings
            </h1>
          </div>
        </div>
        <div
          class="settings-item-toggle"
        >
          <span
            class="settings-item-toggle__label"
          >
            Dark Mode
          </span>
          <label
            class="settings-item-toggle__switch"
          >
            <input
              class="settings-item-toggle__checkbox"
              type="checkbox"
            />
            <span
              class="settings-item-toggle__slider"
            />
          </label>
        </div>
        <fieldset
          class="setting-font-size"
        >
          <legend
            class="setting-font-size__header"
          >
            Font Size
          </legend>
          <input
            checked=""
            class="setting-font-size__input"
            name="font-size"
            type="radio"
          />
          <label
            class="setting-font-size__label"
            for="small-font"
          >
            Small
          </label>
          <input
            class="setting-font-size__input"
            name="font-size"
            type="radio"
          />
          <label
            class="setting-font-size__label"
            for="medium-font"
          >
            Medium
          </label>
          <input
            class="setting-font-size__input"
            name="font-size"
            type="radio"
          />
          <label
            class="setting-font-size__label"
            for="large-font"
          >
            Large
          </label>
        </fieldset>
      </div>
    </div>
    <nav
      class="navbar"
    >
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/event-schedule"
        >
          <svg
            height="36px"
            viewBox="0 0 16 16"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2z"
            />
            <path
              d="M12 3v10h-1V3h1z"
              fill-rule="evenodd"
            />
            <path
              d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1V2.82z"
            />
            <path
              d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5zm0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5zm0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="navbar__title"
          >
            SCHEDULE
          </span>
        </a>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/home"
        >
          <svg
            height="36px"
            viewBox="0 0 32 32"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 16 2.59375 L 15.28125 3.28125 L 2.28125 16.28125 L 3.71875 17.71875 L 5 16.4375 L 5 28 L 14 28 L 14 18 L 18 18 L 18 28 L 27 28 L 27 16.4375 L 28.28125 17.71875 L 29.71875 16.28125 L 16.71875 3.28125 Z M 16 5.4375 L 25 14.4375 L 25 26 L 20 26 L 20 16 L 12 16 L 12 26 L 7 26 L 7 14.4375 Z"
            />
          </svg>
          <span
            class="navbar__title"
          >
            HOME
          </span>
        </a>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/our-story"
        >
          <svg
            height="36px"
            viewBox="0 0 50 50"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 25 2 C 12.309295 2 2 12.309295 2 25 C 2 37.690705 12.309295 48 25 48 C 37.690705 48 48 37.690705 48 25 C 48 12.309295 37.690705 2 25 2 z M 25 4 C 36.609824 4 46 13.390176 46 25 C 46 36.609824 36.609824 46 25 46 C 13.390176 46 4 36.609824 4 25 C 4 13.390176 13.390176 4 25 4 z M 25 11 A 3 3 0 0 0 22 14 A 3 3 0 0 0 25 17 A 3 3 0 0 0 28 14 A 3 3 0 0 0 25 11 z M 21 21 L 21 23 L 22 23 L 23 23 L 23 36 L 22 36 L 21 36 L 21 38 L 22 38 L 23 38 L 27 38 L 28 38 L 29 38 L 29 36 L 28 36 L 27 36 L 27 21 L 26 21 L 22 21 L 21 21 z"
            />
          </svg>
          <span
            class="navbar__title"
          >
            ABOUT
          </span>
        </a>
      </div>
      <div
        class="navbar__content"
      >
        <a
          class="navbar__nav-link "
          href="/undefined/settings"
        >
          <svg
            height="36"
            viewBox="0 0 36 36"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.1001 11C14.2001 11 11.1001 14.1 11.1001 18C11.1001 21.9 14.2001 25 18.1001 25C22.0001 25 25.1001 21.9 25.1001 18C25.1001 14.1 22.0001 11 18.1001 11ZM18.1001 23C15.3001 23 13.1001 20.8 13.1001 18C13.1001 15.2 15.3001 13 18.1001 13C20.9001 13 23.1001 15.2 23.1001 18C23.1001 20.8 20.9001 23 18.1001 23Z"
            />
            <path
              d="M32.8 14.7L30 13.8L29.4 12.3L30.8 9.7C31.1 9.1 31 8.3 30.5 7.8L28.1 5.4C27.6 4.9 26.8 4.8 26.2 5.1L23.6 6.5L22.1 5.9L21.2 3.1C21 2.5 20.4 2 19.7 2H16.3C15.6 2 15 2.5 14.9 3.2L14 6C13.4 6.1 12.9 6.3 12.4 6.6L9.8 5.2C9.2 4.9 8.4 5 7.9 5.5L5.5 7.9C5 8.4 4.9 9.2 5.2 9.8L6.5 12.3C6.3 12.8 6.1 13.4 5.9 13.9L3.1 14.8C2.5 15 2 15.6 2 16.3V19.7C2 20.4 2.5 21 3.2 21.2L6 22.1L6.6 23.6L5.2 26.2C4.9 26.8 5 27.6 5.5 28.1L7.9 30.5C8.4 31 9.2 31.1 9.8 30.8L12.4 29.4L13.9 30L14.8 32.9C15 33.5 15.6 34 16.3 34H19.7C20.4 34 21 33.5 21.2 32.9L22.1 30L23.6 29.4L26.2 30.8C26.8 31.1 27.6 31 28.1 30.5L30.5 28.1C31 27.6 31.1 26.8 30.8 26.2L29.4 23.6L30 22.1L32.9 21.2C33.5 21 34 20.4 34 19.7V16.3C34 15.6 33.5 14.9 32.8 14.7V14.7ZM32 19.4L28.4 20.5L28.3 21L27.4 23.1L27.1 23.6L28.9 26.9L26.9 28.9L23.6 27.1L23.1 27.4C22.4 27.8 21.7 28.1 21 28.3L20.5 28.4L19.4 32H16.6L15.5 28.4L15 28.3L12.9 27.4L12.4 27.1L9.1 28.9L7.1 26.9L8.9 23.6L8.6 23.1C8.2 22.4 7.9 21.7 7.7 21L7.6 20.5L4 19.4V16.6L7.4 15.6L7.6 15.1C7.8 14.3 8.1 13.6 8.5 12.9L8.8 12.4L7.1 9.1L9.1 7.1L12.3 8.9L12.8 8.6C13.5 8.2 14.2 7.9 15 7.7L15.5 7.5L16.6 4H19.4L20.5 7.5L21 7.7C21.7 7.9 22.4 8.2 23.1 8.6L23.6 8.9L26.9 7.1L28.9 9.1L27.1 12.4L27.4 12.9C27.8 13.6 28.1 14.3 28.3 15L28.4 15.5L32 16.6V19.4V19.4Z"
            />
          </svg>
          <span
            class="navbar__title"
          >
            SETTINGS
          </span>
        </a>
      </div>
    </nav>
  </div>
</div>
`;
